File: n.c
#include <stdio.h>
#include <math.h>

int main() {
    int i;
    int count = 0;
    while ( count != 10001 ) {
        i++;
        if ( is_prime( i )) {
            count++;
        }
    }
    printf( "%d\n", i ); 
    return 0;
}

int is_prime( int n ) {
    int d = 2;
    while (1) {
        if ( n % d == 0 ) {
            return 0;
        }
        if ( d < sqrt(n) ) {
            d++; 
        } else {
            return 1;
        }
    }
}

File: n.py
#!/usr/bin/python

import math

def is_prime( n ):
    d = 2
    while 1:
        if n % d == 0:
            return 0
        if d < math.sqrt( n ):
            d = d+1
        else:
            return 1

count = 0
i = 0
while count != 10001:
    i = i+1
    if is_prime( i ):
        count = count+1

print i
